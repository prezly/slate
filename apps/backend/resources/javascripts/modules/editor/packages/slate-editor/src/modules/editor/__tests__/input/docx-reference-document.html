

<div style='mso-element:para-border-div;border:none;border-bottom:solid #4F81BD 1.0pt;
mso-border-bottom-themecolor:accent1;padding:0in 0in 4.0pt 0in'>

<p class=MsoTitle>Title</p>

</div>

<p class=MsoSubtitle>Subtitle</p>

<h1>Heading 1</h1>

<h2>Heading 2</h2>

<h3>Heading 3</h3>

<h4>Heading 4</h4>

<h5>Heading 5</h5>

<h6>Heading 6</h6>

<p class=MsoHeading7>Heading 7</p>

<p class=MsoHeading8>Heading 8</p>

<p class=MsoHeading9>Heading 9</p>

<p class=MsoNormal><b>Bold</b> Normal text <i>Italics </i><u>Underline <b><i>Bold
italics underline</i></b></u> superscript<sup>2 </sup>subscript<sub>2 </sub>normal
text, <a href="https://prezly.com"><span style='text-decoration:none;
text-underline:none'>prezly.com</span></a>,</p>

<p class=MsoNormal><span style='mso-no-proof:yes'><!--[if gte vml 1]><v:shapetype
 id="_x0000_t75" coordsize="21600,21600" o:spt="75" o:preferrelative="t"
 path="m@4@5l@4@11@9@11@9@5xe" filled="f" stroked="f">
 <v:stroke joinstyle="miter"/>
 <v:formulas>
  <v:f eqn="if lineDrawn pixelLineWidth 0"/>
  <v:f eqn="sum @0 1 0"/>
  <v:f eqn="sum 0 0 @1"/>
  <v:f eqn="prod @2 1 2"/>
  <v:f eqn="prod @3 21600 pixelWidth"/>
  <v:f eqn="prod @3 21600 pixelHeight"/>
  <v:f eqn="sum @0 0 1"/>
  <v:f eqn="prod @6 1 2"/>
  <v:f eqn="prod @7 21600 pixelWidth"/>
  <v:f eqn="sum @8 21600 0"/>
  <v:f eqn="prod @7 21600 pixelHeight"/>
  <v:f eqn="sum @10 21600 0"/>
 </v:formulas>
 <v:path o:extrusionok="f" gradientshapeok="t" o:connecttype="rect"/>
 <o:lock v:ext="edit" aspectratio="t"/>
</v:shapetype><v:shape id="Picture_x0020_2" o:spid="_x0000_i1026" type="#_x0000_t75"
 alt="landscape-870x260.jpg" style='width:468pt;height:139.5pt;visibility:visible;
 mso-wrap-style:square'>
 <v:imagedata src="file:///C:\users\kamil\Temp\msohtmlclip1\01\clip_image001.jpg"
  o:title="landscape-870x260"/>
</v:shape><![endif]--><![if !vml]><img border=0 width=624 height=186
src="file:///C:\users\kamil\Temp\msohtmlclip1\01\clip_image002.jpg"
alt=landscape-870x260.jpg v:shapes="Picture_x0020_2"><![endif]></span></p>

<p class=MsoNormal>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed
do eiusmod tempor incididunt ut labore et dolore magna aliqua. Feugiat pretium
nibh ipsum consequat nisl vel pretium. Viverra mauris in aliquam sem fringilla
ut morbi tincidunt augue. Proin libero nunc consequat interdum varius sit amet
mattis vulputate. Et netus et malesuada fames ac turpis. Odio facilisis mauris
sit amet massa. At risus viverra adipiscing at in. Amet luctus venenatis lectus
magna. Sagittis aliquam malesuada bibendum arcu vitae elementum curabitur vitae
nunc. Massa placerat duis ultricies lacus sed turpis. Blandit massa enim nec
dui nunc mattis enim ut tellus.</p>

<p class=MsoNormal><a href="https://example.com/"><span style='color:windowtext;
mso-no-proof:yes;text-decoration:none;text-underline:none'><!--[if gte vml 1]><v:shape
 id="Picture_x0020_3" o:spid="_x0000_i1025" type="#_x0000_t75" alt="dog-300x300.jpeg"
 href="https://example.com/" style='width:225pt;height:225pt;visibility:visible;
 mso-wrap-style:square' o:button="t">
 <v:fill o:detectmouseclick="t"/>
 <v:imagedata src="file:///C:\users\kamil\Temp\msohtmlclip1\01\clip_image003.jpg"
  o:title="dog-300x300"/>
</v:shape><![endif]--><![if !vml]><span style='mso-ignore:vglayout'><img
border=0 width=300 height=300
src="file:///C:\users\kamil\Temp\msohtmlclip1\01\clip_image003.jpg"
alt=dog-300x300.jpeg v:shapes="Picture_x0020_3"></span><![endif]></span></a></p>

<p class=MsoNormal style='margin-bottom:0in;margin-bottom:.0001pt;line-height:
normal'><!--[if gte vml 1]><v:shapetype id="_x0000_t32" coordsize="21600,21600"
 o:spt="32" o:oned="t" path="m,l21600,21600e" filled="f">
 <v:path arrowok="t" fillok="f" o:connecttype="none"/>
 <o:lock v:ext="edit" shapetype="t"/>
</v:shapetype><v:shape id="_x0000_s1026" type="#_x0000_t32" style='position:absolute;
 margin-left:.75pt;margin-top:7.5pt;width:452.25pt;height:0;z-index:1'
 o:connectortype="straight"/><![endif]--><![if !vml]><span style='mso-ignore:
vglayout'>

<table cellpadding=0 cellspacing=0 align=left>
 <tr>
  <td width=0 height=9></td>
 </tr>
 <tr>
  <td></td>
  <td><img width=605 height=2
  src="file:///C:\users\kamil\Temp\msohtmlclip1\01\clip_image004.gif" v:shapes="_x0000_s1026"></td>
 </tr>
</table>

</span><![endif]><span style='font-size:12.0pt;font-family:"Times New Roman","serif";
mso-fareast-font-family:"Times New Roman";mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<br style='mso-ignore:vglayout' clear=ALL>

<p class=MsoListParagraphCxSpFirst style='text-indent:-.25in;mso-list:l0 level1 lfo1'><![if !supportLists]><span
style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>·<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]>Nested List A</p>

<p class=MsoListParagraphCxSpMiddle style='margin-left:1.0in;mso-add-space:
auto;text-indent:-.25in;mso-list:l2 level2 lfo3'><![if !supportLists]><span
style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>·<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]>Nested Lists A1</p>

<p class=MsoListParagraphCxSpMiddle style='margin-left:1.5in;mso-add-space:
auto;text-indent:-.25in;mso-list:l1 level3 lfo2'><![if !supportLists]><span
style='mso-bidi-font-family:Calibri;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>1.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]>Nested Lists A1a</p>

<p class=MsoListParagraphCxSpMiddle style='margin-left:1.5in;mso-add-space:
auto;text-indent:-.25in;mso-list:l1 level3 lfo2'><![if !supportLists]><span
style='mso-bidi-font-family:Calibri;mso-bidi-theme-font:minor-latin'><span
style='mso-list:Ignore'>2.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]>Nested Lists A1b</p>

<p class=MsoListParagraphCxSpMiddle style='margin-left:1.0in;mso-add-space:
auto;text-indent:-.25in;mso-list:l2 level2 lfo3'><![if !supportLists]><span
style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>·<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]>Nested Lists A2</p>

<p class=MsoListParagraphCxSpMiddle style='margin-left:1.5in;mso-add-space:
auto;text-indent:-.25in;mso-list:l0 level3 lfo1'><![if !supportLists]><span
style='font-family:Wingdings;mso-fareast-font-family:Wingdings;mso-bidi-font-family:
Wingdings'><span style='mso-list:Ignore'>§<span style='font:7.0pt "Times New Roman"'>&nbsp;
</span></span></span><![endif]>Nested Lists A2a</p>

<p class=MsoListParagraphCxSpMiddle style='margin-left:1.5in;mso-add-space:
auto;text-indent:-.25in;mso-list:l0 level3 lfo1'><![if !supportLists]><span
style='font-family:Wingdings;mso-fareast-font-family:Wingdings;mso-bidi-font-family:
Wingdings'><span style='mso-list:Ignore'>§<span style='font:7.0pt "Times New Roman"'>&nbsp;
</span></span></span><![endif]>Nested Lists A2b</p>

<p class=MsoListParagraphCxSpMiddle style='margin-left:1.0in;mso-add-space:
auto;text-indent:-.25in;mso-list:l2 level2 lfo3'><![if !supportLists]><span
style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>·<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]>Nested Lists A3</p>

<p class=MsoListParagraphCxSpMiddle style='text-indent:-.25in;mso-list:l0 level1 lfo1'><![if !supportLists]><span
style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>·<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]>Nested Lists B</p>

<p class=MsoListParagraphCxSpMiddle style='margin-left:1.0in;mso-add-space:
auto;text-indent:-.25in;mso-list:l2 level2 lfo3'><![if !supportLists]><span
style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>·<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]>Nested <span style='mso-tab-count:1'>                </span>Lists
B1</p>

<p class=MsoListParagraphCxSpMiddle style='margin-left:1.5in;mso-add-space:
auto;text-indent:-.25in;mso-list:l0 level3 lfo1'><![if !supportLists]><span
style='font-family:Wingdings;mso-fareast-font-family:Wingdings;mso-bidi-font-family:
Wingdings'><span style='mso-list:Ignore'>§<span style='font:7.0pt "Times New Roman"'>&nbsp;
</span></span></span><![endif]>Nested Lists B1a</p>

<p class=MsoListParagraphCxSpMiddle style='margin-left:1.5in;mso-add-space:
auto;text-indent:-.25in;mso-list:l0 level3 lfo1'><![if !supportLists]><span
style='font-family:Wingdings;mso-fareast-font-family:Wingdings;mso-bidi-font-family:
Wingdings'><span style='mso-list:Ignore'>§<span style='font:7.0pt "Times New Roman"'>&nbsp;
</span></span></span><![endif]>Nested Lists B1b</p>

<p class=MsoListParagraphCxSpMiddle style='margin-left:1.0in;mso-add-space:
auto;text-indent:-.25in;mso-list:l2 level2 lfo3'><![if !supportLists]><span
style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>·<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]>Nested Lists B2</p>

<p class=MsoListParagraphCxSpLast style='text-indent:-.25in;mso-list:l0 level1 lfo1'><![if !supportLists]><span
style='font-family:Symbol;mso-fareast-font-family:Symbol;mso-bidi-font-family:
Symbol'><span style='mso-list:Ignore'>·<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]>Nested Lists C</p>

<p class=MsoQuote>This is a quote</p>

<p class=MsoNormal><em><span style='font-family:"Calibri","sans-serif";
mso-ascii-theme-font:minor-latin;mso-hansi-theme-font:minor-latin'>More text</span></em><em><span
style='font-family:"Calibri","sans-serif";mso-ascii-theme-font:minor-latin;
mso-hansi-theme-font:minor-latin;font-style:normal;mso-bidi-font-style:italic'><o:p></o:p></span></em></p>

<p class=MsoNormal><em><span style='font-family:"Calibri","sans-serif";
mso-ascii-theme-font:minor-latin;mso-hansi-theme-font:minor-latin;font-style:
normal;mso-bidi-font-style:italic'><o:p>&nbsp;</o:p></span></em></p>

