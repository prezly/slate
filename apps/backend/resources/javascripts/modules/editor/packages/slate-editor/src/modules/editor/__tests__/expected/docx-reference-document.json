[
    {
        "type": "paragraph",
        "children": [
            {
                "text": "Title"
            }
        ]
    },
    {
        "type": "paragraph",
        "children": [
            {
                "text": "Subtitle"
            }
        ]
    },
    {
        "type": "heading-one",
        "children": [
            {
                "text": "Heading 1"
            }
        ]
    },
    {
        "type": "heading-two",
        "children": [
            {
                "text": "Heading 2"
            }
        ]
    },
    {
        "type": "heading-two",
        "children": [
            {
                "text": "Heading 3"
            }
        ]
    },
    {
        "type": "heading-two",
        "children": [
            {
                "text": "Heading 4"
            }
        ]
    },
    {
        "type": "heading-two",
        "children": [
            {
                "text": "Heading 5"
            }
        ]
    },
    {
        "type": "heading-two",
        "children": [
            {
                "text": "Heading 6"
            }
        ]
    },
    {
        "type": "paragraph",
        "children": [
            {
                "text": "Heading 7"
            }
        ]
    },
    {
        "type": "paragraph",
        "children": [
            {
                "text": "Heading 8"
            }
        ]
    },
    {
        "type": "paragraph",
        "children": [
            {
                "text": "Heading 9"
            }
        ]
    },
    {
        "type": "paragraph",
        "children": [
            {
                "text": "Bold",
                "bold": true
            },
            {
                "text": " Normal text "
            },
            {
                "text": "Italics ",
                "italic": true
            },
            {
                "text": "Underline ",
                "underlined": true
            },
            {
                "text": "Bold italics underline",
                "italic": true,
                "bold": true,
                "underlined": true
            },
            {
                "text": " superscript"
            },
            {
                "text": "2 ",
                "superscript": true
            },
            {
                "text": "subscript"
            },
            {
                "text": "2 ",
                "subscript": true
            },
            {
                "text": "normal text, "
            },
            {
                "href": "https://prezly.com/",
                "type": "link",
                "children": [
                    {
                        "text": "prezly.com"
                    }
                ]
            },
            {
                "text": ","
            }
        ]
    },
    {
        "type": "placeholder:image",
        "children": [
            {
                "text": ""
            }
        ]
    },
    {
        "type": "paragraph",
        "children": [
            {
                "text": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Feugiat pretium nibh ipsum consequat nisl vel pretium. Viverra mauris in aliquam sem fringilla ut morbi tincidunt augue. Proin libero nunc consequat interdum varius sit amet mattis vulputate. Et netus et malesuada fames ac turpis. Odio facilisis mauris sit amet massa. At risus viverra adipiscing at in. Amet luctus venenatis lectus magna. Sagittis aliquam malesuada bibendum arcu vitae elementum curabitur vitae nunc. Massa placerat duis ultricies lacus sed turpis. Blandit massa enim nec dui nunc mattis enim ut tellus."
            }
        ]
    },
    {
        "type": "placeholder:image",
        "children": [
            {
                "text": ""
            }
        ]
    },
    {
        "type": "paragraph",
        "children": [
            {
                "text": "      "
            }
        ]
    },
    {
        "type": "bulleted-list",
        "children": [
            {
                "type": "list-item",
                "children": [
                    {
                        "children": [
                            {
                                "text": "Nested List A"
                            }
                        ],
                        "type": "list-item-text"
                    },
                    {
                        "type": "bulleted-list",
                        "children": [
                            {
                                "type": "list-item",
                                "children": [
                                    {
                                        "children": [
                                            {
                                                "text": "Nested Lists A1"
                                            }
                                        ],
                                        "type": "list-item-text"
                                    },
                                    {
                                        "type": "numbered-list",
                                        "children": [
                                            {
                                                "type": "list-item",
                                                "children": [
                                                    {
                                                        "children": [
                                                            {
                                                                "text": "Nested Lists A1a"
                                                            }
                                                        ],
                                                        "type": "list-item-text"
                                                    }
                                                ]
                                            },
                                            {
                                                "type": "list-item",
                                                "children": [
                                                    {
                                                        "children": [
                                                            {
                                                                "text": "Nested Lists A1b"
                                                            }
                                                        ],
                                                        "type": "list-item-text"
                                                    }
                                                ]
                                            }
                                        ]
                                    }
                                ]
                            },
                            {
                                "type": "list-item",
                                "children": [
                                    {
                                        "children": [
                                            {
                                                "text": "Nested Lists A2"
                                            }
                                        ],
                                        "type": "list-item-text"
                                    },
                                    {
                                        "type": "bulleted-list",
                                        "children": [
                                            {
                                                "type": "list-item",
                                                "children": [
                                                    {
                                                        "children": [
                                                            {
                                                                "text": "Nested Lists A2a"
                                                            }
                                                        ],
                                                        "type": "list-item-text"
                                                    }
                                                ]
                                            },
                                            {
                                                "type": "list-item",
                                                "children": [
                                                    {
                                                        "children": [
                                                            {
                                                                "text": "Nested Lists A2b"
                                                            }
                                                        ],
                                                        "type": "list-item-text"
                                                    }
                                                ]
                                            }
                                        ]
                                    }
                                ]
                            },
                            {
                                "type": "list-item",
                                "children": [
                                    {
                                        "children": [
                                            {
                                                "text": "Nested Lists A3"
                                            }
                                        ],
                                        "type": "list-item-text"
                                    }
                                ]
                            }
                        ]
                    }
                ]
            },
            {
                "type": "list-item",
                "children": [
                    {
                        "children": [
                            {
                                "text": "Nested Lists B"
                            }
                        ],
                        "type": "list-item-text"
                    },
                    {
                        "type": "bulleted-list",
                        "children": [
                            {
                                "type": "list-item",
                                "children": [
                                    {
                                        "children": [
                                            {
                                                "text": "Nested  Lists B1"
                                            }
                                        ],
                                        "type": "list-item-text"
                                    },
                                    {
                                        "type": "bulleted-list",
                                        "children": [
                                            {
                                                "type": "list-item",
                                                "children": [
                                                    {
                                                        "children": [
                                                            {
                                                                "text": "Nested Lists B1a"
                                                            }
                                                        ],
                                                        "type": "list-item-text"
                                                    }
                                                ]
                                            },
                                            {
                                                "type": "list-item",
                                                "children": [
                                                    {
                                                        "children": [
                                                            {
                                                                "text": "Nested Lists B1b"
                                                            }
                                                        ],
                                                        "type": "list-item-text"
                                                    }
                                                ]
                                            }
                                        ]
                                    }
                                ]
                            },
                            {
                                "type": "list-item",
                                "children": [
                                    {
                                        "children": [
                                            {
                                                "text": "Nested Lists B2"
                                            }
                                        ],
                                        "type": "list-item-text"
                                    }
                                ]
                            }
                        ]
                    }
                ]
            },
            {
                "type": "list-item",
                "children": [
                    {
                        "children": [
                            {
                                "text": "Nested Lists C"
                            }
                        ],
                        "type": "list-item-text"
                    }
                ]
            }
        ]
    },
    {
        "type": "block-quote",
        "children": [
            {
                "text": "This is a quote"
            }
        ]
    },
    {
        "type": "paragraph",
        "children": [
            {
                "text": "More text",
                "italic": true
            }
        ]
    },
    {
        "type": "paragraph",
        "children": [
            {
                "text": ""
            }
        ]
    }
]
